bauble/src/value/mod.rs

```diff
@@ -685,13 +687,39 @@ pub(crate) fn resolve_delayed(
                 }
             } && let Some((ty, _)) = &r.asset
             {
+                if let Some(desired_ty) = &d.expected_ty_path {
+                    let span = desired_ty.span;
+                    if let Some(desired_ty) = match &desired_ty.value {
+                        PathKind::Direct(path) => ctx.get_ref(path.borrow()),
+                        PathKind::Indirect(path, ident) => {
+                            ctx.ref_with_ident(path.borrow(), ident.borrow())
+                        }
+                    } && let Some(desired_ty) = desired_ty.ty
+                    {
+                        if desired_ty != *ty {
+                            errors.push(
+                                ConversionError::ExpectedExactType {
+                                    expected: desired_ty,
+                                    got: Some(*ty),
+                                }
+                                .spanned(span),
+                            );
+                        }
+                    } else {
+                        return true;
```

Comment:
Resolving more assets types is not going to change the result of `get_ref`/`ref_with_ident` in terms of finding a type (rather than another asset). Right? So seems like if this fails there will be an infinite loop and this should be an error instead? Also, would it make sense to resolve this to a type ID earlier?

Reply:
In that case this would already produce an error. If the vector is not shrunk then automatically it will produce an error. Also, the reason I did not resolve the type id earlier is because at the point it makes sense to do that the type might not be known.